Snowflake.Data/Core/SFResultSetArrow.cs

```diff
@@ -0,0 +1,176 @@
+                {
+                    // TODO - support for multiple chunks
+                    throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+                }
+
+                responseData.rowSet = null;
+
+                sfResultSetMetaData = new SFResultSetMetaData(responseData);
+
+                isClosed = false;
+
+                queryId = responseData.queryId;
+            }
+            catch(System.Exception ex)
+            {
+                s_logger.Error("Result set error queryId="+responseData.queryId, ex);
+                throw;
+            }
+        }
+
+        internal override async Task<bool> NextAsync()
+        {
+            if (isClosed)
+            {
+                throw new SnowflakeDbException(SFError.DATA_READER_ALREADY_CLOSED);
+            }
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_chunkDownloader != null)
+            {
+                throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+            }
+
+            return false;
+        }
+
+        internal override bool Next()
+        {
+            if (isClosed)
+            {
+                throw new SnowflakeDbException(SFError.DATA_READER_ALREADY_CLOSED);
+            }
+
+            _currentChunkRowIdx++;
+            if (_currentChunkRowIdx < _currentChunkRowCount)
+            {
+                return true;
+            }
+
+            if (_chunkDownloader != null)
+            {
+                throw new SnowflakeDbException(SFError.UNSUPPORTED_FEATURE);
+            }
+
+            return false;
+        }
+
+        internal override bool NextResult()
+        {
+            return false;
+        }
+
+        internal override async Task<bool> NextResultAsync(CancellationToken cancellationToken)
+        {
+            return await Task.FromResult(false);
+        }
+
+        internal override bool HasRows()
+        {
+            if (isClosed)
+            {
+                return false;
+            }
+
+            return _currentChunkRowCount > 0 || _totalChunkCount > 0;
+        }
+
+        /// <summary>
+        /// Move cursor back one row.
+        /// </summary>
+        /// <returns>True if it works, false otherwise.</returns>
+        internal override bool Rewind()
+        {
+            if (isClosed)
+            {
+                throw new SnowflakeDbException(SFError.DATA_READER_ALREADY_CLOSED);
+            }
+
+            if (_currentChunkRowIdx >= 0)
+            {
+                _currentChunkRowIdx--;
+                if (_currentChunkRowIdx >= _currentChunkRowCount)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        internal override UTF8Buffer getObjectInternal(int columnIndex)
+        {
+            if (isClosed)
```

Comment:
could we extract `isClosed` validatation with excetion thrown to private method and use in all other methods where it is needed?